src/Features/Onboarding.php

```diff
@@ -120,6 +133,26 @@ private function add_actions() {
 		add_action( 'current_screen', array( $this, 'redirect_old_onboarding' ) );
 	}
 
+	/**
+	 * Trigger the woocommerce_onboarding_profile_completed action
+	 *
+	 * @param array $old_value Previous value.
+	 * @param array $value Current value.
+	 */
+	public function trigger_profile_completed_action( $old_value, $value ) {
+		if ( ! isset( $value['completed'] ) || ! $value['completed'] ) {
+			return;
+		}
+
+		/**
+		 * Action hook fired when the onboarding profile (or onboarding wizard,
+		 * or profiler) is completed.
+		 *
+		 * @since 1.5.0
+		 */
+		do_action( 'woocommerce_onboarding_profile_completed' );
```

Comment:
This is a really nice addition!  I can see a couple areas where it would be useful to replace with this action instead of manually checking the profile data.  We might want to create a follow-up issue to update those.

Reply:
I've created a follow-up issue: https://github.com/woocommerce/woocommerce-admin/issues/4931